client/components/wpadmin-auto-login/index.jsx

```diff
@@ -0,0 +1,65 @@
+/**
+ * External Dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { get } from 'lodash';
+
+export default class WpadminAutoLogin extends Component {
+
+	static propTypes = {
+		site: PropTypes.object,
+		delay: PropTypes.number,
+	}
+
+	state = {
+		timeout: null,
+		retries: 10,
+	};
+
+	static defaultProps = {
+		delay: 3000,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.onerror = this.onerror.bind( this );
+		this.requestLogin = this.requestLogin.bind( this );
+	}
+
+	componentWillMount() {
+		this.state.timeout = setTimeout( this.requestLogin, this.props.delay );
+	}
+
+	getPixelUrl( siteUrl ) {
+		const pixel = encodeURI( siteUrl + '/wp-includes/images/blank.gif' );
+		return `${ siteUrl }/wp-login.php?redirect_to=${ pixel }`;
+	}
+
+	/**
+	 * If we encounter an error, try again.
+	 * We may want to retry since just post-transfer there are certificate issues.
+	 */
+	onerror() {
```

Comment:
Since we're just binding things in the constructor, we can avoid it altogether and instead use class properties, like you did in this class with `state`:

```jsx
onerror = () => {
   //...
};
```